.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,115 @@
+      - name: Install dependencies
+        run: |
+          pip3 install --user toml
+          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
+      - name: Run setup install
+        run:  python3 setup.py install --user
+      - name: Test install
+        run:  python3 -c "import precice"
+        
+  setup_install_single_version_externally_managed:
+    name: Run setup install --single-version-externally-managed (for spack)
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
+      - name: Install dependencies
+        run: |
+          pip3 install --user toml
+          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
+      - name: Run setup install --single-version-externally-managed
+        run:  python3 setup.py install --single-version-externally-managed --root=/
+        
+  setup_test:
+    name: Run setup test
+    runs-on: ubuntu-latest
+    container: precice/ci-ubuntu-1804
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get update && apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
+      - name: Checkout precice and make required files discoverable
+        run: | 
+          git clone https://github.com/precice/precice.git precice-core
+          mkdir -p precice
+          cp precice-core/src/precice/SolverInterface.hpp precice/SolverInterface.hpp
+      - name: Install dependencies
+        run: |
+          pip3 install --user toml
+          python3 -c 'import toml; c = toml.load("pyproject.toml"); print("\n".join(c["build-system"]["requires"]))' | pip3 install -r /dev/stdin
+      - name: Run setup test
+        run:  python3 setup.py test
+        
+  pip_install:
+    name: Run pip install
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
+      - name: Run pip install
+        run:  pip3 install --user .
+      - name: Run pip install
+        run:  python3 -c "import precice"
+
+  solverdummy_test:
+    name: Run solverdummy
+    runs-on: ubuntu-latest
+    container: benjaminrueth/precice
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Install & upgrade pip3
+        run: |
+          apt-get install -y python3-pip
+          pip3 install --upgrade --user pip
```

Comment:
Is it considered safe to mix the `pip` version distributed via the Ubuntu repository with a user-installed and updated version? I remember having troubles with that some time in the past (or a similar setup).